modules/tinymce/src/plugins/template/test/ts/browser/TemplateSanityTest.ts

```diff
@@ -78,4 +75,65 @@ describe('browser.tinymce.plugins.template.TemplateSanityTest', () => {
     editor.execCommand('mceInsertTemplate', false, '<p>{$first+name}</p>');
     TinyAssertions.assertContent(editor, '<p>Tester</p>');
   });
+
+  it('TINY-9244: Sanitized html should be shown when previewing template', async () => {
+    const editor = hook.editor();
+    const unsanitizedHtml = '<img src="error" onerror="throw new Error();">';
+    addSettings({
+      templates: [{ title: 'a', description: 'b', content: unsanitizedHtml }],
+    });
+    const unsanitizedPreviewHtmlSelector = 'p > img[src="error"][onerror="throw new Error();"]';
+    const sanitizedPreviewHtmlSelector = 'p > img[src="error"][data-mce-src="error"]';
+    const assertPreviewContent = (dialogEl: SugarElement<Node>, existsSelector: string, notExistsSelector: string): void => {
+      UiFinder.findIn<HTMLIFrameElement>(dialogEl, 'iframe').fold(
+        () => assert.fail('Preview iframe not found'),
+        (iframe) => {
+          // fallback for pre-IE 8 using contentWindow.document
+          const iframeDoc = iframe.dom.contentDocument || iframe.dom.contentWindow?.document;
+          const iframeBody = SugarElement.fromDom(iframeDoc?.body as Node);
+          UiFinder.exists(iframeBody, existsSelector);
+          UiFinder.notExists(iframeBody, notExistsSelector);
+        }
+      );
+    };
+
+    try {
+      await pPreviewTemplate(editor, (dialogEl) => assertPreviewContent(dialogEl, sanitizedPreviewHtmlSelector, unsanitizedPreviewHtmlSelector));
+    } catch {
+      assert.fail('Unsanitized html read');
+    }
+  });
+
+  const assertFnDoesNotReadUnsanitizedHtmlInDom = async (editor: Editor, fn: (unsanitizedHtml: string) => void | Promise<void>): Promise<void> => {
+    const fnReadsUnsanitizedHtmlInDom = async () => {
+      const unsanitizedHtml = '<img src="error" onerror="window.document.unsanitizedHtmlFn();">';
+      let isUnsanitizedHtmlRead = false;
+      (editor.getDoc() as any).unsanitizedHtmlFn = () => {
+        isUnsanitizedHtmlRead = true;
+      };
+      await fn(unsanitizedHtml);
+      // wait for unsanitized html to be read and error to be thrown
+      await Waiter.pWait(1000);
```

Comment:
Change this to a waiter. Delaying here for 1 second seems wasteful.

Reply:
No wait this is one of those wait for something not to happen the worst case. :cry:
I wonder is it super important to have this test I don't think we have similar tests for the parser for example that could in theory do similar things.

Reply:
This function is used by 2 only tests that verifies that unsanitized HTML is not read during template insertion, so my concern is that our testing may not catch any potential regressions if I remove these 2 tests. On the plus side, I find that as long as the `pWait` time is non-zero, ex: 1ms, the function is reliable enough to catch any error thrown by the unsanitized HTML. Is `pWait(1)` good enough?